pkg/ebpf/events_pipeline.go

```diff
@@ -555,6 +561,55 @@ func (t *Tracee) getStackAddresses(stackID uint32) ([]uint64, error) {
 	return stackAddresses[0:stackCounter], nil
 }
 
+func (t *Tracee) enrichProcess(ctx context.Context, in <-chan *trace.Event) (<-chan *trace.Event, <-chan error) {
+	out := make(chan *trace.Event)
+	errc := make(chan error, 1)
+
+	go func() {
+		defer close(out)
+		defer close(errc)
+
+		for {
+			select {
+			case event := <-in:
+				currentProcess, err := t.processTree.GetProcessInfo(event.HostProcessID, event.Timestamp)
```

Comment:
We can get only partial information here, as events order is not guaranteed and the exec event might come after current one.
In the container enrichment, we wait for the information about the container to be fulfilled and only then pass the event onward.
We should consider if we want the same behavior here as well.

Reply:
@rafaeldtinoco WDYT, since we worked on the container enrichment together.
There is a key difference here where the latency of the network call for enrichment doesn't exist, BUT, process tree processing may not finish in time for enrichment here which is sort of similar (although we would likely reduce the chance of this by moving the processing to the process stage which is earlier in the pipeline).

Reply:
So I resolve this as this will be solved when we move the control plane events